packages/flame/lib/src/game/game_widget/game_widget.dart

```diff
@@ -8,122 +8,38 @@ import 'package:flame/src/game/game_widget/gesture_detector_builder.dart';
 import 'package:flutter/services.dart';
 import 'package:flutter/widgets.dart';
 
-typedef GameLoadingWidgetBuilder = Widget Function(
-  BuildContext,
-);
-
-typedef GameErrorWidgetBuilder = Widget Function(
-  BuildContext,
-  Object error,
-);
-
-typedef OverlayWidgetBuilder<T extends Game> = Widget Function(
-  BuildContext context,
-  T game,
-);
-
-typedef GameFactory<T extends Game> = T Function();
-
-/// A [StatefulWidget] that is in charge of attaching a [Game] instance into the
-/// Flutter tree.
+/// The **GameWidget** is a Flutter widget which is used to insert a [Game]
+/// instance into the Flutter widget tree.
+///
+/// The `GameWidget` is sufficiently feature-rich to run as the root of your
+/// Flutter application. Thus, the simplest way to use `GameWidget` is like
+/// this:
+/// ```dart
+/// void main() {
+///   runApp(
+///     GameWidget(game: MyGame()),
+///   );
+/// }
+/// ```
+///
+/// At the same time, `GameWidget` is a regular Flutter widget, and can be
+/// inserted arbitrarily deep into the widget tree, including the possibility
+/// of having multiple `GameWidget`s within a single app.
+///
+/// The layout behavior of this widget is that it will expand to fill all
+/// available space. Thus, when used as a root widget it will make the app
+/// full-screen. Inside any other layout widget it will take as much space as
```

Comment:
One thing that I think we should also mention here, that `GameWidget` doesn't clip automatically. Meaning that is the user render something outside the boundaries of the parent of the game widget, the game content will "overflow", that if the user wants to clip, they should wrap the game widget into a ClipRect widget.

Reply:
Good point. Will need to add clipping to Flame Studio as well.

Reply:
Done.